packages/amplify-schema-validator/src/index.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+/* eslint-disable global-require */
+
+import { parse } from 'graphql';
+import { validateIndexScalarTypes } from './validators/index-scalar-types';
+import { validateRequireBelongsToRelation } from './validators/relation-required-with-belongs-to';
+import { validateManyToManyTwoLocations } from './validators/two-many-to-many-locations';
+import { ValidationError } from './exceptions/validation-error';
+
+const allValidators = [
+  validateIndexScalarTypes,
+  validateRequireBelongsToRelation,
+  validateManyToManyTwoLocations,
+];
+
+/**
+ * The primary export of this library
+ * runs all validators and throws a ValidationException with all failure reasons
+ *
+ * @param schemaString the graphql schema
+ * @returns void
+ */
+export const validateSchema = (schemaString: string): void => {
+  const schema = parse(schemaString);
+  let aggregatedErrors: Error[] = [];
+  allValidators.forEach((validate) => {
+    const errors = validate(schema);
+    aggregatedErrors = aggregatedErrors.concat(errors);
+  });
```

Comment:
Rather than relying on mutable state here, I'd prefer just `flagMapping` over validators.

e.g. ```const validationErrors = allValidators.flatMap(validate => validate(schema));```